src/bullsAndCows.js

```diff
@@ -26,7 +26,31 @@
  * @return {object} - like {bulls: 0, cows: 0}
  */
 function bullsAndCows(generatedNumber, enteredNumber) {
-  // write code here
+  if (enteredNumber === undefined
+    || enteredNumber.length !== 4
+    || !/[0-9]/g.test(enteredNumber)
+    || enteredNumber.match(/(.)\1+/g)) {
+    return undefined;
+  }
+
+  const entered = [...enteredNumber];
+  const generated = [...generatedNumber];
+  const result = {
+    bulls: 0,
+    cows: 0,
+  };
+
+  for (let i = 0; i < 4; i++) {
+    if (generated[i] === entered[i]) {
+      result.bulls++;
+    } else {
+      if (entered.find(item => generated[i] === item) !== undefined) {
```

Comment:
зачем else а потом if, если можно `else if {}`
для простоты можно было использовать `includes`:
```suggestion
      if (entered.includes(generated[i])) {
```